File: src/SOP8/Base/Base.Model/History/ComponentDetail.cs
using System;
using dnsDapperDBUtil.DataAccessLayer.IDAL;

namespace Base.Model.History
{
	public class ComponentDetail : Table
	{
		public enum Fields { compn_hist_detail_sn, compn_hist_sn, data_no, data_intgr, data_float, data_str };
		public enum WriteFields { compn_hist_sn, data_no, data_intgr, data_float, data_str };

		public int compn_hist_detail_sn { get; set; }
		public int compn_hist_sn { get; set; }
		public int data_no { get; set; }
		public int? data_intgr { get; set; }
		public double? data_float { get; set; }
		public string/* nullable */ data_str { get; set; }

		public static string TableName { get { return "his_compn_detail"; } }

		public override string GetTableName()
		{
			return TableName;
		}

		public override string GetPrimaryCondition()
		{
			return string.Format("{0} = {1}", Fields.compn_hist_detail_sn, compn_hist_detail_sn);
		}

		public override Type GetFieldType()
		{
			return typeof(Fields);
		}

		public override Type GetWriteFieldType()
		{
			return typeof(WriteFields);
		}

		public void FromCopy(ComponentDetail obj)
		{
			this.compn_hist_detail_sn = obj.compn_hist_detail_sn;
			this.compn_hist_sn = obj.compn_hist_sn;
			this.data_no = obj.data_no;
			this.data_intgr = obj.data_intgr;
			this.data_float = obj.data_float;
			this.data_str = obj.data_str;
		}
	}
}

File: src/SOP8/Egovframework/Base/base-model/src/main/java/egovframework/base/model/history/ComponentDetail.java
package egovframework.base.model.history;

import lombok.*;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ComponentDetail {
	public enum Fields { compn_hist_detail_sn, compn_hist_sn, data_no, data_intgr, data_float, data_str }

	private int compn_hist_detail_sn;

	private int compn_hist_sn;

	private int data_no;

	private Integer data_intgr;

	private Double data_float;

	private String data_str;

	public static final String autoIncreaseField = "compn_hist_detail_sn";
	public static final String tableName = "his_compn_detail";
	public static final String[] primaryKeys = { "compn_hist_detail_sn" };

}
